go-controller/pkg/ovn/zone_interconnect/zone_ic_handler.go

```diff
@@ -633,7 +636,8 @@ func (zic *ZoneInterconnectHandler) addRemoteNodeStaticRoutes(node *corev1.Node,
 	addRoute := func(prefix, nexthop string) error {
 		logicalRouterStaticRoute := nbdb.LogicalRouterStaticRoute{
 			ExternalIDs: map[string]string{
-				"ic-node": node.Name,
+				"ic-node":               node.Name,
+				types.NetworkExternalID: zic.GetNetworkName(),
```

Comment:
IIRC we usually also add the `types.TopologyExternalID`, it wouldn't hurt to have it here also, especially if we allow same network names on different topologies.